packages/web-frontend/src/components/View/MatrixWrapper/components/FooterRow.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { getMarkerForOption } from '../../../Marker';
+import { LEGEND_COLOR_ICON } from '../../../Marker/markerIcons';
+import { CONDITIONAL_MATRIX_CONDITION_SHAPE } from '../../propTypes';
+
+const Legend = ({ legend, styles }) => {
+  return (
+    <div style={styles.footer.legend}>
+      {legend.map(legendItem => {
+        return <LegendItem {...legendItem} key={legendItem.legendLabel} styles={styles} />;
```

Comment:
Sorry, I'm still not completely familiar with react, is this `key` a special property that's needed here, and does it require unique values?

Reply:
Ah yep, React wants one for every "list" of elements so it can re-render faster - https://reactjs.org/docs/lists-and-keys.html#keys